packages/react-next/src/components/SwatchColorPicker/SwatchColorPicker.base.tsx

```diff
@@ -16,361 +8,325 @@ import {
+   */
+  const onGridCellHovered = React.useCallback(
+    (item?: IColorCellProps): void => {
+      if (onCellHovered) {
+        return item ? onCellHovered(item.id, item.color) : onCellHovered();
       }
-    }
-  };
+    },
+    [onCellHovered],
+  );
 
   /**
-   * Callback to make sure we don't update the hovered element during mouse wheel
+   * Callback passed to the GridCell class that will trigger the onCellFocus callback of the SwatchColorPicker
    */
-  private _onWheel = (): void => {
-    this._setNavigationTimeout();
-  };
+  const onGridCellFocused = React.useCallback(
+    (item?: IColorCellProps): void => {
+      if (onCellFocused) {
+        if (item) {
+          internalState.cellFocused = true;
+          return onCellFocused(item.id, item.color);
+        } else {
+          internalState.cellFocused = false;
+          return onCellFocused();
+        }
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [onCellFocused],
+  );
 
   /**
-   * Callback that
+   * Handle the click on a cell
    */
-  private _onKeyDown = (ev: React.KeyboardEvent<HTMLButtonElement>): void => {
-    if (
-      ev.which === KeyCodes.up ||
-      ev.which === KeyCodes.down ||
-      ev.which === KeyCodes.left ||
-      ev.which === KeyCodes.right
-    ) {
-      this._setNavigationTimeout();
-    }
-  };
+  const onCellClick = React.useCallback(
+    (item: IColorCellProps): void => {
+      if (disabled) {
+        return;
+      }
+
+      if (item.id !== selectedId) {
+        if (onCellFocused && internalState.cellFocused) {
+          internalState.cellFocused = false;
+          onCellFocused();
+        }
+        setSelectedId(item.id);
+      }
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [disabled, onCellFocused, selectedId],
+  );
 
   /**
    * Sets a timeout so we won't process any mouse "hover" events
    * while navigating (via mouseWheel or arrowKeys)
    */
-  private _setNavigationTimeout = () => {
-    if (!this.isNavigationIdle && this.navigationIdleTimeoutId !== undefined) {
-      this.async.clearTimeout(this.navigationIdleTimeoutId);
-      this.navigationIdleTimeoutId = undefined;
+  const setNavigationTimeout = React.useCallback(() => {
+    if (!internalState.isNavigationIdle && internalState.navigationIdleTimeoutId !== undefined) {
+      clearTimeout(internalState.navigationIdleTimeoutId);
+      internalState.navigationIdleTimeoutId = undefined;
     } else {
-      this.isNavigationIdle = false;
+      internalState.isNavigationIdle = false;
     }
 
-    this.navigationIdleTimeoutId = this.async.setTimeout(() => {
-      this.isNavigationIdle = true;
-    }, this.navigationIdleDelay);
-  };
+    internalState.navigationIdleTimeoutId = setTimeout(() => {
+      internalState.isNavigationIdle = true;
+    }, internalState.navigationIdleDelay);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
Even though in this case we know none of the things currently referenced will change, it's still safer to just add them to deps so the rule can stay enabled to catch potential future mistakes.
```suggestion
  }, [internalState, setTimeout, clearTimeout]);
```